### PR TITLE
Bug 1169030 - Update Gallery to use modern L10n API

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1169030

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ef3379662a5666e57d93a739b10b7911d2bad6db

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=83d1a3b976fd47b11fe1838e1ab9315d9109ec9a

File: apps/gallery/js/thumbnail_item.js [hunk 1]
```diff
@@ -32,17 +32,17 @@ function ThumbnailItem(fileData) {
...
   var date = new Date(this.data.date);
-  var description = navigator.mozL10n.get(!this.data.metadata.video ?
-    'imageDescriptionShort' : 'videoDescriptionShort');
-  var label = ThumbnailItem.formatter.localeFormat(date, description);
-  this.imgNode.setAttribute('aria-label', label);
+  var descId = !this.data.metadata.video ?
+    'imageDescriptionShort' : 'videoDescriptionShort';
+  navigator.mozL10n.formatValue(descId, function(description) {
+    var label = ThumbnailItem.formatter.localeFormat(date, description);
+    this.imgNode.setAttribute('aria-label', label);
```
Comment (punamdahiya) [inline apps/gallery/js/thumbnail_item.js]: Tested and the aria-label attributes are not getting set

File: apps/gallery/js/thumbnail_date_group.js [hunk 2]
```diff
@@ -139,9 +134,11 @@ ThumbnailDateGroup.formatter = new navigator.mozL10n.DateTimeFormat();
 
 ThumbnailDateGroup.prototype.localize = function() {
   var date = new Date(this.date);
-  var format = navigator.mozL10n.get('date-group-header');
-  var formattedDate = ThumbnailDateGroup.formatter.localeFormat(date, format);
-  this.header.textContent = formattedDate;
+  navigator.mozL10n.formatValue('date-group-header', function(format) {
+    var formattedDate =
+      ThumbnailDateGroup.formatter.localeFormat(date, format);
+    this.header.textContent = formattedDate;
```
Comment (punamdahiya) [inline apps/gallery/js/thumbnail_date_group.js]: This code is not reachable as on testing gallery app with attached patch, group header text in thumbnail list is not getting set.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=690fdf89f6eb439df2fa0429aa42d146244e9a43

File: apps/gallery/js/thumbnail_date_group.js (hunk 2 above)
Comment (zbraniecki) [inline apps/gallery/js/thumbnail_date_group.js]: agh, I forgot that we switched formatValue to be a Promise. Fixed in the updated patch :)

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=70ce5f6460c967a7bc850bea0864038acd4a73a2

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=72c7eb77b73b3571160f85e06268e2f68037cb02

File: apps/gallery/js/thumbnails.js [hunk 3]
```diff
@@ -18,13 +18,7 @@
...
   thumbnails.list = new ThumbnailList(ThumbnailDateGroup, thumbnails.container);
 
-  // Localize dates and item desc in thumbnails list when the locale changes
-  // Bug 1135256: Having a listener in ThumbnailList prevents them from
-  // being garbage collected since ready() holds a strong reference, so
-  // we have a singleton listener here.
-  navigator.mozL10n.ready(function localize() {
-    thumbnails.list.localize();
-  });
+  navigator.mozL10n.ready(thumbnails.list.localize.bind(thumbnails.list));
```
Comment (punamdahiya) [inline apps/gallery/js/thumbnails.js]: nit: I understand even with l10n API, Bug 1135256 is still there and we should keep comment from Line 21-24 explaining why we are not moving this listener to ThumbnailList

File: apps/gallery/js/startup.js [hunk 4]
```diff
@@ -6,7 +6,7 @@
 /* global files, thumbnailClickHandler */
 
 // When the document is loaded and localized, we can start running
-navigator.mozL10n.once(function startup() {
```
Comment (punamdahiya) [inline apps/gallery/js/startup.js]: nit: We can delete comment in Line 8

File: apps/gallery/js/thumbnails.js (hunk 3 above)
Comment (zbraniecki) [inline apps/gallery/js/thumbnails.js]: good point!

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ce0af3f4a480eabd503ff9125c9de9761d20851

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#zHhNVfQqThiQx9DX9QUeXQ

The pull request failed to pass integration tests. It could not be landed, please try again.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a18fc213a0ec92b5e40a13c36c0263b5976be4bc

**Comment by autolander** [comment]
http://docs.taskcluster.net/tools/task-graph-inspector/#-lup7HNORiSncrv9Z-17_w

The pull request failed to pass integration tests. It could not be landed, please try again.

**Pull request merged**